src/scales/scale.time.js

```diff
@@ -106,32 +159,151 @@ function interpolate(table, skey, sval, tkey) {
 	return prev[tkey] + offset;
 }
 
+/**
+ * Convert the given value to a moment object using the given time options.
+ * @see http://momentjs.com/docs/#/parsing/
+ */
+function momentify(value, options) {
+	var parser = options.parser;
+	var format = options.parser || options.format;
+
+	if (typeof parser === 'function') {
+		return parser(value);
+	}
+
+	if (typeof value === 'string' && typeof format === 'string') {
+		return moment(value, format);
+	}
+
+	if (!(value instanceof moment)) {
+		value = moment(value);
+	}
+
+	if (value.isValid()) {
+		return value;
+	}
+
+	// Labels are in an incompatible moment format and no `parser` has been provided.
+	// The user might still use the deprecated `format` option to convert his inputs.
+	if (typeof format === 'function') {
+		return format(value);
+	}
+
+	return value;
+}
+
 function parse(input, scale) {
 	if (helpers.isNullOrUndef(input)) {
 		return null;
 	}
 
-	var round = scale.options.time.round;
-	var value = scale.getRightValue(input);
-	var time = value.isValid ? value : helpers.time.parseTime(scale, value);
-	if (!time || !time.isValid()) {
+	var options = scale.options.time;
+	var value = momentify(scale.getRightValue(input), options);
+	if (!value.isValid()) {
 		return null;
 	}
 
-	if (round) {
-		time.startOf(round);
+	if (options.round) {
+		value.startOf(options.round);
 	}
 
-	return time.valueOf();
+	return value.valueOf();
 }
 
-module.exports = function(Chart) {
+function determineStepSize(min, max, unit, capacity) {
+	var range = max - min;
+	var interval = INTERVALS[unit];
+	var milliseconds = interval.size;
+	var steps = interval.steps;
+	var i, ilen, factor;
+
+	if (!steps) {
+		return Math.ceil(range / ((capacity || 1) * milliseconds));
```

Comment:
It took me a while to figure out the correct formula :) Not sure what extra test we can add since now this only happens for `unit: 'year'` and I think there is already many test using years.